cpu/efm32/periph/i2c.c

```diff
@@ -34,7 +34,14 @@
 
 static volatile I2C_TransferReturn_TypeDef i2c_progress[I2C_NUMOF];
 
-static mutex_t i2c_lock[I2C_NUMOF];
+/**
+ * @brief   Initialized bus locks (we have a maximum of three devices)
+ */
+static mutex_t i2c_lock[] = {
+    MUTEX_INIT,
+    MUTEX_INIT,
+    MUTEX_INIT
+};
```

Comment:
Since this is hard coded at 3 instances, we should also check that `I2C_NUMOF <= 3`

Add
```c
#if I2C_NUMOF > 3
#error too many I2C instances, update i2c.c
#endif
```
or similar

Reply:
Hmm, this isn't going to work,:

```
../periph_conf.h:36:40: error: missing binary operator before token "("
 #define PERIPH_NUMOF(config)    (sizeof(config) / sizeof(config[0]))
```

The number of devices is asserted in `i2c_init_master`. Will that be enough for now?

Reply:
Should be 
```c
#define PERIPH_NUMOF    (sizeof(config) / sizeof(config[0]))
```
?

Reply:
I didn't think of that. An assert in i2c_init should be enough then.

Reply:
I have remove that last commit, so we just assert the number of devices in the `i2c_init_master`, which was already the case.

Reply:
For reference, the assert is [here](https://github.com/RIOT-OS/RIOT/blob/fb8ade23313268c2c60f7933798c2dfa8dc57458/cpu/efm32/periph/i2c.c#L67).

Reply:
I was thinking adding an assert for the mutex array size:
```c
assert(I2C_NUMOF <= (sizeof(i2c_lock) / sizeof(i2c_lock[0]));
```
The condition outcome will be precomputed during compilation, but it will prevent hard to find bugs if someone configures too many buses.
(this could have been made a static assert if this was C++)

Reply:
Right, I'll add this in the `i2c_init_master`.